frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,299 @@
+        <h2>{t('public~Creating resources...')}</h2>
+      </>
+    );
+  } else if (!sending && !errors) {
+    StatusBlock = (
+      <>
+        <GreenCheckCircleIcon size="lg" />
+        <h2>{t('public~Resources successfully created')}</h2>
+      </>
+    );
+  } else {
+    StatusBlock = (
+      <>
+        <YellowExclamationTriangleIcon size="lg" />
+        <h2>{t('public~One or more resources failed to create')}</h2>
+      </>
+    );
+  }
+  return <div className="co-import-yaml-status text-center">{StatusBlock}</div>;
+};
+
+const ImportYAMLResourceStatus: React.FC<ImportYAMLResourceStatusProps> = ({ status }) => {
+  let StatusIcon;
+  if (status.creating) {
+    StatusIcon = <Spinner size="sm" className="co-icon-space-r" />;
+  } else if (status.error) {
+    StatusIcon = <RedExclamationCircleIcon className="co-icon-space-r" size="sm" />;
+  } else {
+    StatusIcon = <GreenCheckCircleIcon className="co-icon-space-r" size="sm" />;
+  }
+  return (
+    <span>
+      {StatusIcon}
+      {status.message}
+    </span>
+  );
+};
+
+const IMPORT_COLS = {
+  NAME: 0,
+  NAMESPACE: 1,
+  STATUS: 2,
+};
+const COLUMN_MAP = ['metadata.name', 'metadata.namespace', 'message'];
+
+const getTableCell = (row, column) => {
+  return column === IMPORT_COLS.STATUS
+    ? _.get(row.status, COLUMN_MAP[column])
+    : _.get(row.obj, COLUMN_MAP[column]);
+};
+
+const ImportYAMLResults: React.FC<ImportYAMLResultsPageProps> = ({
+  activeNamespace,
+  location,
+  models,
+}) => {
+  const { t } = useTranslation();
+  const [rows, setRows] = React.useState(
+    location.state.objs.map((obj) => ({
+      obj: { ...obj },
+      status: { creating: true, message: t('public~Creating') },
+    })),
+  );
+  const [sending, setSending] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState(-1);
+  const [activeSortDirection, setActiveSortDirection] = React.useState(null);
+  const [ready, setReady] = React.useState(false);
+  React.useEffect(() => {
+    // Add a ready state so we're not rendering a flicker of a stale table (with the browser back button)
+    // before the history.goBack action occurs
+    if (location.state.done) {
+      return history.goBack();
+    }
+    setReady(true);
+    const requests = createRequests(location.state.objs, models);
+    Promise.allSettled(requests).then(() => {
+      history.replace({ state: { ...location.state, done: true } });
+      setSending(false);
+    });
+    requests.forEach((resourceRequest, index) => {
+      resourceRequest
+        .then((resourceObj) =>
+          setRows((prevRows) =>
+            // eslint-disable-next-line max-nested-callbacks
+            prevRows.map((resourceRow) =>
+              resourceObj?.metadata?.name === resourceRow.obj.metadata.name &&
+              resourceObj?.kind === resourceRow.obj.kind &&
+              resourceObj?.metadata?.namespace === resourceRow.obj.metadata.namespace
```

Comment:
This is not guaranteed to be unique since we can have resources with the same kind in different API groups.

Instead of storing rows in state and trying to update them, can we simply generate the correct rows on render? I think this would be simpler.

Reply:
changed the way to index the data. we are generating the rows on render, but maybe the naming of the state was implying otherwise. updated the state names for better readability as well.